svm-transaction/src/svm_transaction.rs

```diff
@@ -0,0 +1,27 @@
+use {
+    crate::SVMMessage,
+    solana_sdk::{hash::Hash, signature::Signature, transaction::VersionedTransaction},
+};
+
+mod reference;
+mod sanitized_transaction;
+
+pub trait SVMTransaction: SVMMessage {
+    /// Get the first signature of the message.
+    fn signature(&self) -> &Signature;
+
+    /// Get all the signatures of the message.
+    fn signatures(&self) -> &[Signature];
+
+    /// Returns the message hash.
+    // TODO: consider moving this to Message
+    fn message_hash(&self) -> &Hash;
+
+    /// Returns true if the transaction is a simple vote transaction.
+    // TODO: consider moving this to Message
+    fn is_simple_vote_transaction(&self) -> bool;
```

Comment:
Based on what you mention in the PR description...

> @Lichtso had mentioned to me that he felt we use "transaction" incorrectly right now, and a "transaction" is a loaded message. i.e. users always send messages, transactions are what come out of those messages

...I like the idea of sweeping these into message. You'd probably need a message accessor in here, though. Something like this maybe?

```rust
fn message<M: SVMMessage>(&self) -> &M;
```

Reply:
Actually I very specifically do not want to have message accessors here.
New type I'm planning will just implement both these traits.

Reply:
Correct me if I'm wrong @Lichtso, but I think he meant that just everything we have here should probably be along lines of `Message` and `SignedMessage (the one with signature)` for the traits rather than `Transaction` as the name

Reply:
yeah basically kept the separate traits so that our current functions that don't require us to have signature information don't have to be rewritten.

A lot of the functions further down the callchain just take `SanitizedMessage`, and I'd rather not break all those tests by having a single trait